test/python/transpiler/test_echo_rzx_weyl_decomposition.py

```diff
@@ -0,0 +1,202 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Test the EchoRZXWeylDecomposition pass and the TwoQubitWeylEchoRZX class"""
+import unittest
+from math import pi
+import numpy as np
+
+from qiskit import QuantumRegister, QuantumCircuit
+
+from qiskit.transpiler.passes.optimization.echo_rzx_weyl_decomposition import (
+    EchoRZXWeylDecomposition,
+)
+from qiskit.converters import circuit_to_dag, dag_to_circuit
+from qiskit.test import QiskitTestCase
+from qiskit.test.mock import FakeParis
+
+import qiskit.quantum_info as qi
+
+from qiskit.quantum_info.synthesis.two_qubit_decompose import (
+    TwoQubitWeylDecomposition,
+    TwoQubitWeylEchoRZX,
+)
+
+
+class TestEchoRZXWeylDecomposition(QiskitTestCase):
+    """Tests the EchoRZXWeylDecomposition pass and the TwoQubitWeylEchoRZX class."""
+
+    def setUp(self):
+        super().setUp()
+        self.backend = FakeParis()
+        self.inst_map = self.backend.defaults().instruction_schedule_map
+
+    def test_native_weyl_decomposition(self):
```

Comment:
Would it make sense to test the number hadamard gates added in the circuit to check that it properly picks up on the native VS non-native directions?

Reply:
See 4139093.